src/Diagnostics/dump_state_and_aux.jl

```diff
@@ -0,0 +1,41 @@
+using ..Mesh.Interpolation
+
+using Printf
+
+function dump_state_and_aux_init(diagrp, currtime)
+    @assert diagrp.interpol !== nothing
+end
+
+function dump_state_and_aux_collect(diagrp, currtime)
+    DA = CLIMA.array_type()
+
+    dg = Settings.dg
+    Q  = Settings.Q
+    FT = eltype(Q.data)
+    bl = dg.balancelaw
+
+    istate = DA(Array{FT}(undef, diagrp.interpol.Npl, num_state(bl, FT)))
+    iaux   = DA(Array{FT}(undef, diagrp.interpol.Npl, num_aux(bl, FT)))
+
+    # interpolate and save
+    interpolate_local!(diagrp.interpol, Q.data,           istate, project = diagrp.project)
+    interpolate_local!(diagrp.interpol, dg.auxstate.data, iaux,   project = diagrp.project)
+
+    # filename (may also want to take out)
+    nprefix = @sprintf("%s_%s-%f-step%04d",
+                       diagrp.out_prefix, diagrp.name, Settings.starttime, diagrp.step)
+    statefilename = joinpath(Settings.output_dir, "$(nprefix).nc")
+    auxfilename = joinpath(Settings.output_dir, "$(nprefix)_aux.nc")
+
+    #statenames = flattenednames(vars_state(bl, FT))
+    statenames = ("rho", "rhou", "rhov", "rhow", "rhoe")
```

Comment:
Can you add a method for the user to overwrite the state names if necessary and leave `flattenednames(vars_state(bl, FT))` as the default? This way `dump_state_and_aux` would be generic to any model by default.

Reply:
I've switched to using the `flattenednames` for now. These files will have to be post-processed anyway, so it should be possible to work with the Unicode. If not, I'll figure out a way to allow overriding the names.